camel/models/base_model.py

```diff
@@ -20,9 +20,9 @@
 from camel.types import (
     ChatCompletion,
     ChatCompletionChunk,
+    InnerModelType,
```

Comment:
Why not just use `ModelType` here?

Reply:
As per our discussion, we will keep `ModelType` immutable since it's a Enum type. Therefore we have this `InnerModelType` to support the compatibility of both `ModelType` and `str`, and since `ModelType` and `InnerModelType` all inherits `str`, and `ModelType` in the same time is a subclass of `InnerModelType`, then can be used as `str` seamlessly, and `InnerModelType` should be only used explicitly in `ModelBackEnd` and `ModelFactory`.

Reply:
@Wendong-Fan After discussing with Guohao we have decided to rename `InnerModelType` into `UnifiedModelType` to make it easier to understand. I'll do this after this PR has been merged so I'll mark it as resolved here